src/lib/rest/rest.cpp

```diff
@@ -616,6 +630,17 @@ static int contentTypeCheck(ConnectionInfo* ciP)
     return 1;
   }
 
+
+  // Case 5
+  if ((ciP->apiVersion == "v2") && (ciP->httpHeaders.contentType != "application/json"))
```

Comment:
True. We should do something there also, about accepted output.
In rest.cpp there is a function (outFormatCheck) doing just this, but it accepts JSON and XML.
If the request is v2 then we should only accept JSON ...

Tricky ...
XML is still the default format.
If we change that ... things may stop working for some people ...

However, this PR is big enough already; let's not add to it ...
Create as issue, pls